nomenclature/core.py

```diff
@@ -24,6 +27,9 @@ def __init__(self, path):
             "region", path / "regions", top_level_attr="hierarchy"
         )
 
+        if not self.region or not self.variable:
+            logger.warning("Attribute `variable` or `region` is empty.")
```

Comment:
For more specific warnings... (not sure if this works, only "dry-coding")
```suggestion
        args = [(self.region, "region"), (self.variable, "variable")]
        for attr, name in args:
            if attr.empty:
                logger.warning(f"Attribute {name} is empty.")
```